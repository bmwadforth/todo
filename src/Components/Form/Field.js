import React from "react";
import PropTypes from "prop-types";

export default function Field(props) {
    return (
        <div className="form-item">
            <input id={props.id} type={props.type ? props.type : "text"} onChange={e => {
                props.onChange(e);
            }} value={props.value} placeholder={props.title} required={props.required}/>
        </div>
    );
}

Field.propTypes = {
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
    value: PropTypes.any.isRequired,
    required: PropTypes.bool.isRequired,
    type: PropTypes.string
};
