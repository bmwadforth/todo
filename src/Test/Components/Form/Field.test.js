import React from 'react';
import {shallow, configure} from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import Field from "../../../Components/Form/Field";
configure({adapter: new Adapter()});

describe("Field Component", () => {
    it("Should Match Snapshot", () => {
        const wrapper = shallow(<Field id="Test" title="Test" onChange={() => {}} required/>);
        expect(wrapper.html()).toMatchSnapshot();
    });
});
