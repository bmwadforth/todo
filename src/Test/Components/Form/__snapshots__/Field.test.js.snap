// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field Component Should Match Snapshot 1`] = `"<div class=\\"form-item\\"><input type=\\"text\\" id=\\"Test\\" placeholder=\\"Test\\" required=\\"\\"/></div>"`;
